0.8.42 breaks build

We see build errors in Slint with the 0.8.42 release:

```
error[E0599]: the method `as_bytes` exists for reference `&SharedPixelBuffer<Rgb<u8>>`, but its trait bounds were not satisfied
  --> internal/core/graphics/image/cache.rs:18:59
   |
18 |                 SharedImageBuffer::RGB8(pixels) => pixels.as_bytes().len(),
   |                                                           ^^^^^^^^ method cannot be called on `&SharedPixelBuffer<Rgb<u8>>` due to unsatisfied trait bounds
   |
note: trait bound `[RGB<u8>]: ComponentBytes<u8>` was not satisfied
  --> internal/core/graphics/image.rs:94:14
   |
92 | impl<Pixel: Clone + rgb::Pod> SharedPixelBuffer<Pixel>
   |                               ------------------------
93 | where
94 |     [Pixel]: rgb::ComponentBytes<u8>,
   |              ^^^^^^^^^^^^^^^^^^^^^^^ unsatisfied trait bound introduced here
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `as_bytes`, perhaps you need to implement one of them:
           candidate #1: `ComponentBytes`
           candidate #2: `EncodableLayout`
           candidate #3: `OsStrExt`
           candidate #4: `rgb::legacy::internal::pixel::ComponentBytes`

```

Bisection suggests that this regressed with commit 788e50fa60247a5c6954944c9319abfe7088ef2d

Since 0.8.42 is a patch release, this unfortunately breaks the build of our existing releases.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,8 +17,6 @@ edition = "2021"
 default = ["as-bytes", "argb", "grb"]
 # safe as_bytes() casts require a marker trait for non-padded, non-pointery types.
 as-bytes = ["bytemuck"]
-# Preferred over as-bytes
-bytemuck = ["dep:bytemuck"]
 # Deprecated: all pixel types are enabled by default. No-op.
 argb = []
 # Deprecated: all pixel types are enabled by default. No-op.

--- a/src/bytemuck.rs
+++ b/src/bytemuck.rs
@@ -1,18 +1,27 @@
+use crate::alt::{Gray, GrayAlpha, BGR, BGRA, GRB};
+use crate::alt::{ABGR, ARGB};
+use crate::ComponentBytes;
 use crate::{RGB, RGBA};
-use crate::alt::{Gray, GrayAlpha, BGR, BGRA};
-use crate::alt::{ARGB, ABGR};
 
 #[cfg(feature = "as-bytes")]
 unsafe impl<T> crate::Pod for RGB<T> where T: crate::Pod {}
 #[cfg(feature = "as-bytes")]
 unsafe impl<T> crate::Pod for BGR<T> where T: crate::Pod {}
 #[cfg(feature = "as-bytes")]
+unsafe impl<T> crate::Pod for GRB<T> where T: crate::Pod {}
+#[cfg(feature = "as-bytes")]
 unsafe impl<T> crate::Zeroable for RGB<T> where T: crate::Zeroable {}
 #[cfg(feature = "as-bytes")]
 unsafe impl<T> crate::Zeroable for BGR<T> where T: crate::Zeroable {}
+#[cfg(feature = "as-bytes")]
+unsafe impl<T> crate::Zeroable for GRB<T> where T: crate::Zeroable {}
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Zeroable for ABGR<T, A> where T: crate::Zeroable, A: crate::Zeroable {
+unsafe impl<T, A> crate::Zeroable for ABGR<T, A>
+where
+    T: crate::Zeroable,
+    A: crate::Zeroable,
+{
     #[track_caller]
     #[inline(always)]
     fn zeroed() -> Self {
@@ -31,13 +40,27 @@ const fn assert_no_padding<T, A, S>() {
 }
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Pod for RGBA<T, A> where T: crate::Pod, A: crate::Pod {}
+unsafe impl<T, A> crate::Pod for RGBA<T, A>
+where
+    T: crate::Pod,
+    A: crate::Pod,
+{
+}
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Pod for BGRA<T, A> where T: crate::Pod, A: crate::Pod {}
+unsafe impl<T, A> crate::Pod for BGRA<T, A>
+where
+    T: crate::Pod,
+    A: crate::Pod,
+{
+}
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Zeroable for RGBA<T, A> where T: crate::Zeroable, A: crate::Zeroable {
+unsafe impl<T, A> crate::Zeroable for RGBA<T, A>
+where
+    T: crate::Zeroable,
+    A: crate::Zeroable,
+{
     #[track_caller]
     #[inline(always)]
     fn zeroed() -> Self {
@@ -49,13 +72,27 @@ unsafe impl<T, A> crate::Zeroable for RGBA<T, A> where T: crate::Zeroable, A: cr
 }
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Pod for ARGB<T, A> where T: crate::Pod, A: crate::Pod {}
+unsafe impl<T, A> crate::Pod for ARGB<T, A>
+where
+    T: crate::Pod,
+    A: crate::Pod,
+{
+}
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Pod for ABGR<T, A> where T: crate::Pod, A: crate::Pod {}
+unsafe impl<T, A> crate::Pod for ABGR<T, A>
+where
+    T: crate::Pod,
+    A: crate::Pod,
+{
+}
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Zeroable for ARGB<T, A> where T: crate::Zeroable, A: crate::Zeroable {
+unsafe impl<T, A> crate::Zeroable for ARGB<T, A>
+where
+    T: crate::Zeroable,
+    A: crate::Zeroable,
+{
     #[track_caller]
     #[inline(always)]
     fn zeroed() -> Self {
@@ -67,7 +104,11 @@ unsafe impl<T, A> crate::Zeroable for ARGB<T, A> where T: crate::Zeroable, A: cr
 }
 
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Zeroable for BGRA<T, A> where T: crate::Zeroable, A: crate::Zeroable {
+unsafe impl<T, A> crate::Zeroable for BGRA<T, A>
+where
+    T: crate::Zeroable,
+    A: crate::Zeroable,
+{
     #[track_caller]
     #[inline(always)]
     fn zeroed() -> Self {
@@ -81,19 +122,59 @@ unsafe impl<T, A> crate::Zeroable for BGRA<T, A> where T: crate::Zeroable, A: cr
 #[cfg(feature = "as-bytes")]
 unsafe impl<T> crate::Pod for Gray<T> where T: crate::Pod {}
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Pod for GrayAlpha<T, A> where T: crate::Pod, A: crate::Pod {}
+unsafe impl<T, A> crate::Pod for GrayAlpha<T, A>
+where
+    T: crate::Pod,
+    A: crate::Pod,
+{
+}
 #[cfg(feature = "as-bytes")]
 unsafe impl<T> crate::Zeroable for Gray<T> where T: crate::Zeroable {}
 #[cfg(feature = "as-bytes")]
-unsafe impl<T, A> crate::Zeroable for GrayAlpha<T, A> where T: crate::Zeroable, A: crate::Zeroable {
+unsafe impl<T, A> crate::Zeroable for GrayAlpha<T, A>
+where
+    T: crate::Zeroable,
+    A: crate::Zeroable,
+{
     #[track_caller]
     #[inline(always)]
     fn zeroed() -> Self {
         unsafe {
-            if core::mem::size_of::<A>() + core::mem::size_of::<T>() != core::mem::size_of::<Self>() {
+            if core::mem::size_of::<A>() + core::mem::size_of::<T>() != core::mem::size_of::<Self>()
+            {
                 panic!("type has padding");
             }
             core::mem::zeroed()
         }
+    }
+}
+
+#[cfg(feature = "as-bytes")]
+impl<T: crate::Pod> ComponentBytes<T> for [Gray<T>] {
+    #[inline]
+    fn as_bytes(&self) -> &[u8] {
+        assert_ne!(0, core::mem::size_of::<T>());
+        ::bytemuck::cast_slice(self)
+    }
+
+    #[inline]
+    fn as_bytes_mut(&mut self) -> &mut [u8] {
+        assert_ne!(0, core::mem::size_of::<T>());
+        ::bytemuck::cast_slice_mut(self)
+    }
+}
+
+#[cfg(feature = "as-bytes")]
+impl<T: crate::Pod> ComponentBytes<T> for [GrayAlpha<T>] {
+    #[inline]
+    fn as_bytes(&self) -> &[u8] {
+        assert_ne!(0, core::mem::size_of::<T>());
+        ::bytemuck::cast_slice(self)
+    }
+
+    #[inline]
+    fn as_bytes_mut(&mut self) -> &mut [u8] {
+        assert_ne!(0, core::mem::size_of::<T>());
+        ::bytemuck::cast_slice_mut(self)
     }
 }

--- a/src/legacy/internal/pixel.rs
+++ b/src/legacy/internal/pixel.rs
@@ -35,27 +35,12 @@ pub trait ComponentSlice<T> {
 ///
 /// Plain types are not allowed to contain struct padding, booleans, chars, enums, references or pointers.
 #[cfg(feature = "as-bytes")]
-pub trait ComponentBytes<T: ::bytemuck::Pod> {
+pub trait ComponentBytes<T: crate::Pod> {
     /// The components interpreted as raw bytes, in machine's native endian. In `RGB` bytes of the red component are first.
     fn as_bytes(&self) -> &[u8];
 
     /// The components interpreted as raw bytes, in machine's native endian. In `RGB` bytes of the red component are first.
     fn as_bytes_mut(&mut self) -> &mut [u8];
-}
-
-#[cfg(feature = "as-bytes")]
-impl<T: ::bytemuck::Pod> ComponentBytes<T> for [T] {
-    #[inline]
-    fn as_bytes(&self) -> &[u8] {
-       assert_ne!(0, core::mem::size_of::<T>());
-       ::bytemuck::cast_slice(self)
-    }
-
-    #[inline]
-    fn as_bytes_mut(&mut self) -> &mut [u8] {
-        assert_ne!(0, core::mem::size_of::<T>());
-        ::bytemuck::cast_slice_mut(self)
-    }
 }
 
 /// Applying operation to every component
@@ -74,7 +59,8 @@ pub trait ComponentMap<DestPixel, SrcComponent, DestComponent> {
     ///
     /// Note that it returns the pixel directly, not an Interator.
     fn map<Callback>(&self, f: Callback) -> DestPixel
-        where Callback: FnMut(SrcComponent) -> DestComponent;
+    where
+        Callback: FnMut(SrcComponent) -> DestComponent;
 }
 
 /// Same as `ComponentMap`, but doesn't change the alpha channel (if there's any alpha).
@@ -83,5 +69,6 @@ pub trait ColorComponentMap<DestPixel, SrcComponent, DestComponent> {
     ///
     /// Note that it returns the pixel directly, not an Interator.
     fn map_c<Callback>(&self, f: Callback) -> DestPixel
-        where Callback: FnMut(SrcComponent) -> DestComponent;
+    where
+        Callback: FnMut(SrcComponent) -> DestComponent;
 }

--- a/src/legacy/internal/rgb.rs
+++ b/src/legacy/internal/rgb.rs
@@ -1,13 +1,16 @@
 use super::pixel::*;
 use crate::alt::GRB;
 use crate::alt::{BGR, BGRA};
+use crate::ComponentSlice;
 use crate::{RGB, RGBA};
 use core::fmt;
 
 impl<T> BGR<T> {
     /// Convenience function for creating a new pixel
     /// Warning: The order of arguments is R,G,B
-    #[deprecated(note = "This function has a misleading order of arguments. Use BGR{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use BGR{} literal instead"
+    )]
     #[inline(always)]
     pub const fn new(r: T, g: T, b: T) -> Self {
         Self { b, g, r }
@@ -27,11 +30,13 @@ macro_rules! impl_rgb {
         impl<T: Copy, B> ComponentMap<$RGB<B>, T, B> for $RGB<T> {
             #[inline(always)]
             fn map<F>(&self, mut f: F) -> $RGB<B>
-                where F: FnMut(T) -> B {
+            where
+                F: FnMut(T) -> B,
+            {
                 $RGB {
-                    r:f(self.r),
-                    g:f(self.g),
-                    b:f(self.b),
+                    r: f(self.r),
+                    g: f(self.g),
+                    b: f(self.b),
                 }
             }
         }
@@ -39,11 +44,13 @@ macro_rules! impl_rgb {
         impl<T: Copy, B> ColorComponentMap<$RGB<B>, T, B> for $RGB<T> {
             #[inline(always)]
             fn map_c<F>(&self, mut f: F) -> $RGB<B>
-                where F: FnMut(T) -> B {
+            where
+                F: FnMut(T) -> B,
+            {
                 $RGB {
-                    r:f(self.r),
-                    g:f(self.g),
-                    b:f(self.b),
+                    r: f(self.r),
+                    g: f(self.g),
+                    b: f(self.b),
                 }
             }
         }
@@ -51,25 +58,19 @@ macro_rules! impl_rgb {
         impl<T> ComponentSlice<T> for $RGB<T> {
             #[inline(always)]
             fn as_slice(&self) -> &[T] {
-                unsafe {
-                    core::slice::from_raw_parts(self as *const Self as *const T, 3)
-                }
+                unsafe { core::slice::from_raw_parts(self as *const Self as *const T, 3) }
             }
 
             #[inline(always)]
             fn as_mut_slice(&mut self) -> &mut [T] {
-                unsafe {
-                    core::slice::from_raw_parts_mut(self as *mut Self as *mut T, 3)
-                }
+                unsafe { core::slice::from_raw_parts_mut(self as *mut Self as *mut T, 3) }
             }
         }
 
         impl<T> ComponentSlice<T> for [$RGB<T>] {
             #[inline]
             fn as_slice(&self) -> &[T] {
-                unsafe {
-                    core::slice::from_raw_parts(self.as_ptr() as *const _, self.len() * 3)
-                }
+                unsafe { core::slice::from_raw_parts(self.as_ptr() as *const _, self.len() * 3) }
             }
 
             #[inline]
@@ -80,6 +81,20 @@ macro_rules! impl_rgb {
             }
         }
 
+        #[cfg(feature = "as-bytes")]
+        impl<T: crate::Pod> ComponentBytes<T> for [$RGB<T>] {
+            #[inline]
+            fn as_bytes(&self) -> &[u8] {
+                assert_ne!(0, core::mem::size_of::<T>());
+                ::bytemuck::cast_slice(self)
+            }
+
+            #[inline]
+            fn as_bytes_mut(&mut self) -> &mut [u8] {
+                assert_ne!(0, core::mem::size_of::<T>());
+                ::bytemuck::cast_slice_mut(self)
+            }
+        }
     };
 }
 
@@ -181,40 +196,42 @@ mod rgb_test {
 
     #[test]
     fn grb_test() {
-        let grb = GRB {g:1,r:2,b:3}.map(|c| c * 2) + 1;
+        let grb = GRB { g: 1, r: 2, b: 3 }.map(|c| c * 2) + 1;
         let rgb: crate::RGB8 = grb.into();
-        assert_eq!(rgb, RGB::new(5,3,7));
+        assert_eq!(rgb, RGB::new(5, 3, 7));
     }
 
     #[test]
     fn sanity_check() {
-        let neg = RGB::new(1,2,3i32).map(|x| -x);
+        let neg = RGB::new(1, 2, 3i32).map(|x| -x);
         assert_eq!(neg.r, -1);
         assert_eq!(neg.g, -2);
         assert_eq!(neg.b, -3);
 
-        let mut px = RGB::new(3,4,5);
+        let mut px = RGB::new(3, 4, 5);
         px.as_mut_slice()[1] = 111;
         assert_eq!(111, px.g);
 
-        assert_eq!(RGBA::new(250,251,252,253), RGB::new(250,251,252).with_alpha(253));
+        assert_eq!(
+            RGBA::new(250, 251, 252, 253),
+            RGB::new(250, 251, 252).with_alpha(253)
+        );
 
-        assert_eq!(RGB{r:1u8,g:2,b:3}, RGB::new(1u8,2,3));
-        assert!(RGB{r:1u8,g:1,b:2} < RGB::new(2,1,1));
+        assert_eq!(RGB { r: 1u8, g: 2, b: 3 }, RGB::new(1u8, 2, 3));
+        assert!(RGB { r: 1u8, g: 1, b: 2 } < RGB::new(2, 1, 1));
 
         let mut h = std::collections::HashSet::new();
         h.insert(px);
-        assert!(h.contains(&RGB::new(3,111,5)));
-        assert!(!h.contains(&RGB::new(111,5,3)));
-
+        assert!(h.contains(&RGB::new(3, 111, 5)));
+        assert!(!h.contains(&RGB::new(111, 5, 3)));
 
         #[cfg(feature = "as-bytes")]
         {
-            let v = vec![RGB::new(1u8,2,3), RGB::new(4,5,6)];
-            assert_eq!(&[1,2,3,4,5,6], v.as_bytes());
+            let v = vec![RGB::new(1u8, 2, 3), RGB::new(4, 5, 6)];
+            assert_eq!(&[1, 2, 3, 4, 5, 6], v.as_bytes());
         }
 
-        assert_eq!(RGB::new(0u8,0,0), Default::default());
+        assert_eq!(RGB::new(0u8, 0, 0), Default::default());
     }
 
     #[test]

--- a/src/legacy/internal/rgba.rs
+++ b/src/legacy/internal/rgba.rs
@@ -16,7 +16,9 @@ impl<T> BGRA<T> {
     #[inline(always)]
     /// Convenience function for creating a new pixel
     /// Warning: The order of arguments is R,G,B,A
-    #[deprecated(note = "This function has a misleading order of arguments. Use BGRA{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use BGRA{} literal instead"
+    )]
     pub const fn new(r: T, g: T, b: T, a: T) -> Self {
         Self { b, g, r, a }
     }
@@ -29,7 +31,9 @@ impl<T, A> BGRA<T, A> {
     #[inline(always)]
     /// Convenience function for creating a new pixel
     /// Warning: The order of arguments is R,G,B,A
-    #[deprecated(note = "This function has a misleading order of arguments. Use BGRA{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use BGRA{} literal instead"
+    )]
     pub const fn new_alpha(r: T, g: T, b: T, a: A) -> Self {
         Self { b, g, r, a }
     }
@@ -39,7 +43,9 @@ impl<T> ARGB<T> {
     #[inline(always)]
     /// Convenience function for creating a new pixel
     /// The order of arguments is R,G,B,A
-    #[deprecated(note = "This function has a misleading order of arguments. Use ARGB{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use ARGB{} literal instead"
+    )]
     pub const fn new(r: T, g: T, b: T, a: T) -> Self {
         Self { r, g, b, a }
     }
@@ -49,7 +55,9 @@ impl<T, A> ARGB<T, A> {
     #[inline(always)]
     /// Convenience function for creating a new pixel
     /// The order of arguments is R,G,B,A
-    #[deprecated(note = "This function has a misleading order of arguments. Use ARGB{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use ARGB{} literal instead"
+    )]
     pub const fn new_alpha(r: T, g: T, b: T, a: A) -> Self {
         Self { r, g, b, a }
     }
@@ -59,7 +67,9 @@ impl<T> ABGR<T> {
     #[inline(always)]
     /// Convenience function for creating a new pixel
     /// The order of arguments is R,G,B,A
-    #[deprecated(note = "This function has a misleading order of arguments. Use ABGR{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use ABGR{} literal instead"
+    )]
     pub const fn new(r: T, g: T, b: T, a: T) -> Self {
         Self { r, g, b, a }
     }
@@ -69,7 +79,9 @@ impl<T, A> ABGR<T, A> {
     #[inline(always)]
     /// Convenience function for creating a new pixel
     /// The order of arguments is R,G,B,A
-    #[deprecated(note = "This function has a misleading order of arguments. Use ABGR{} literal instead")]
+    #[deprecated(
+        note = "This function has a misleading order of arguments. Use ABGR{} literal instead"
+    )]
     pub const fn new_alpha(r: T, g: T, b: T, a: A) -> Self {
         Self { r, g, b, a }
     }
@@ -103,7 +115,10 @@ macro_rules! impl_rgba {
             /// Create new RGBA with the same alpha value, but different RGB values
             #[inline(always)]
             pub fn map_rgb<F, U, B>(&self, mut f: F) -> $RGBA<U, B>
-                where F: FnMut(T) -> U, U: Clone, B: From<A> + Clone
+            where
+                F: FnMut(T) -> U,
+                U: Clone,
+                B: From<A> + Clone,
             {
                 $RGBA {
                     r: f(self.r),
@@ -123,14 +138,21 @@ macro_rules! impl_rgba {
             #[inline(always)]
             /// Create a new RGBA with the new alpha value, but same RGB values
             pub fn with_alpha(&self, a: A) -> Self {
-                Self { r: self.r, g: self.g, b: self.b, a }
+                Self {
+                    r: self.r,
+                    g: self.g,
+                    b: self.b,
+                    a,
+                }
             }
 
             /// Create a new RGBA with a new alpha value created by the callback.
             /// Allows changing of the type used for the alpha channel.
             #[inline]
             pub fn map_alpha<F, B>(&self, f: F) -> $RGBA<T, B>
-                where F: FnOnce(A) -> B {
+            where
+                F: FnOnce(A) -> B,
+            {
                 $RGBA {
                     r: self.r,
                     g: self.g,
@@ -143,7 +165,9 @@ macro_rules! impl_rgba {
         impl<T: Copy, B> ComponentMap<$RGBA<B>, T, B> for $RGBA<T> {
             #[inline(always)]
             fn map<F>(&self, mut f: F) -> $RGBA<B>
-            where F: FnMut(T) -> B {
+            where
+                F: FnMut(T) -> B,
+            {
                 $RGBA {
                     r: f(self.r),
                     g: f(self.g),
@@ -156,7 +180,9 @@ macro_rules! impl_rgba {
         impl<T: Copy, A: Copy, B> ColorComponentMap<$RGBA<B, A>, T, B> for $RGBA<T, A> {
             #[inline(always)]
             fn map_c<F>(&self, mut f: F) -> $RGBA<B, A>
-            where F: FnMut(T) -> B {
+            where
+                F: FnMut(T) -> B,
+            {
                 $RGBA {
                     r: f(self.r),
                     g: f(self.g),
@@ -169,25 +195,19 @@ macro_rules! impl_rgba {
         impl<T> ComponentSlice<T> for $RGBA<T> {
             #[inline(always)]
             fn as_slice(&self) -> &[T] {
-                unsafe {
-                    core::slice::from_raw_parts(self as *const Self as *const T, 4)
-                }
+                unsafe { core::slice::from_raw_parts(self as *const Self as *const T, 4) }
             }
 
             #[inline(always)]
             fn as_mut_slice(&mut self) -> &mut [T] {
-                unsafe {
-                    core::slice::from_raw_parts_mut(self as *mut Self as *mut T, 4)
-                }
+                unsafe { core::slice::from_raw_parts_mut(self as *mut Self as *mut T, 4) }
             }
         }
 
         impl<T> ComponentSlice<T> for [$RGBA<T>] {
             #[inline]
             fn as_slice(&self) -> &[T] {
-                unsafe {
-                    core::slice::from_raw_parts(self.as_ptr() as *const _, self.len() * 4)
-                }
+                unsafe { core::slice::from_raw_parts(self.as_ptr() as *const _, self.len() * 4) }
             }
 
             #[inline]
@@ -198,6 +218,20 @@ macro_rules! impl_rgba {
             }
         }
 
+        #[cfg(feature = "as-bytes")]
+        impl<T: crate::Pod> ComponentBytes<T> for [$RGBA<T>] {
+            #[inline]
+            fn as_bytes(&self) -> &[u8] {
+                assert_ne!(0, core::mem::size_of::<T>());
+                ::bytemuck::cast_slice(self)
+            }
+
+            #[inline]
+            fn as_bytes_mut(&mut self) -> &mut [u8] {
+                assert_ne!(0, core::mem::size_of::<T>());
+                ::bytemuck::cast_slice_mut(self)
+            }
+        }
     };
 }
 
@@ -343,7 +377,7 @@ impl<T: fmt::Display, A: fmt::Display> fmt::Display for BGRA<T, A> {
 
 #[test]
 fn rgba_test() {
-    let neg = RGBA::new(1,2,3i32,1000).map(|x| -x);
+    let neg = RGBA::new(1, 2, 3i32, 1000).map(|x| -x);
     assert_eq!(neg.r, -1);
     assert_eq!(neg.rgb().r, -1);
     assert_eq!(neg.g, -2);
@@ -351,18 +385,23 @@ fn rgba_test() {
     assert_eq!(neg.b, -3);
     assert_eq!(neg.rgb().b, -3);
     assert_eq!(neg.a, -1000);
-    assert_eq!(neg.map_alpha(|x| x+1).a, -999);
+    assert_eq!(neg.map_alpha(|x| x + 1).a, -999);
     assert_eq!(neg, neg.as_slice().iter().copied().collect());
-    assert!(neg < RGBA::new(0,0,0,0));
+    assert!(neg < RGBA::new(0, 0, 0, 0));
 
-    let neg = RGBA::new(1u8,2,3,4).map_rgb(|c| -(c as i16));
+    let neg = RGBA::new(1u8, 2, 3, 4).map_rgb(|c| -(c as i16));
     assert_eq!(-1i16, neg.r);
     assert_eq!(4i16, neg.a);
-    let neg = RGBA::new(1u8,2,3,4).map_c(|c| -(c as i16));
+    let neg = RGBA::new(1u8, 2, 3, 4).map_c(|c| -(c as i16));
     assert_eq!(-1i16, neg.r);
     assert_eq!(4u8, neg.a);
 
-    let mut px = RGBA{r:1,g:2,b:3,a:4};
+    let mut px = RGBA {
+        r: 1,
+        g: 2,
+        b: 3,
+        a: 4,
+    };
     px.as_mut_slice()[3] = 100;
     assert_eq!(1, px.rgb_mut().r);
     assert_eq!(2, px.rgb_mut().g);
@@ -372,15 +411,20 @@ fn rgba_test() {
 
     #[cfg(feature = "as-bytes")]
     {
-        let v = vec![RGBA::new(1u8,2,3,4), RGBA::new(5,6,7,8)];
-        assert_eq!(&[1,2,3,4,5,6,7,8], v.as_bytes());
+        let v = vec![RGBA::new(1u8, 2, 3, 4), RGBA::new(5, 6, 7, 8)];
+        assert_eq!(&[1, 2, 3, 4, 5, 6, 7, 8], v.as_bytes());
     }
 }
 
 #[test]
 #[cfg(feature = "as-bytes")]
 fn abgr_test() {
-    let abgr = ABGR {r:1,g:2,b:3,a:4};
+    let abgr = ABGR {
+        r: 1,
+        g: 2,
+        b: 3,
+        a: 4,
+    };
     assert_eq!(4, abgr.as_slice()[0]);
     use crate::AsPixels;
     assert_eq!(abgr, [abgr].as_bytes().as_pixels()[0]);
@@ -403,7 +447,7 @@ fn bgra_test() {
     assert_eq!(neg.b, -3);
     assert_eq!(neg.bgr().b, -3);
     assert_eq!(neg.a, -1000);
-    assert_eq!(&[-3,-2,-1,-1000], neg.as_slice());
+    assert_eq!(&[-3, -2, -1, -1000], neg.as_slice());
     assert!(neg < BGRA::new(0, 0, 0, 0));
 
     let neg = BGRA::new(1u8, 2u8, 3u8, 4u8).map_rgb(|c| -(c as i16));
@@ -413,7 +457,13 @@ fn bgra_test() {
     assert_eq!(-1i16, neg.r);
     assert_eq!(4u8, neg.a);
 
-    let mut px = BGRA{r:1,g:2,b:3,a:-9}.alpha(4);
+    let mut px = BGRA {
+        r: 1,
+        g: 2,
+        b: 3,
+        a: -9,
+    }
+    .alpha(4);
     px.as_mut_slice()[3] = 100;
     assert_eq!(1, px.bgr_mut().r);
     assert_eq!(2, px.bgr_mut().g);
@@ -421,10 +471,9 @@ fn bgra_test() {
     assert_eq!(4, px.bgr_mut().b);
     assert_eq!(100, px.a);
 
-
     #[cfg(feature = "as-bytes")]
     {
         let v = vec![BGRA::new(3u8, 2, 1, 4), BGRA::new(7, 6, 5, 8)];
-        assert_eq!(&[1,2,3,4,5,6,7,8], v.as_bytes());
+        assert_eq!(&[1, 2, 3, 4, 5, 6, 7, 8], v.as_bytes());
     }
 }

